openedx/core/lib/courses.py

```diff
@@ -43,3 +47,32 @@ def create_course_image_thumbnail(course, dimensions):
     _content, thumb_loc = contentstore().generate_thumbnail(course_image, dimensions=dimensions)
 
     return StaticContent.serialize_asset_key_with_slash(thumb_loc)
+
+
+def clean_course_id(model_form, is_required=True):
+    """
+    Validate the course_id for a Django ModelForm.
```

Comment:
I can improve.  This code was mostly just copied/moved from an existing function that was duplicated many times with minor edits.  Hopefully this becomes the last duplicate, and other locations can be refactored to use this.  I could create a separate PR for that, but didn't want this one to explode.